feat: migrate react-intl to lingui

rajat-ws opened the pull request:
### Ticket Link

---

### Related Links

---

### Description

- Migration of `react-intl` to `lingui`
- Reduced bundle size
- Updated UI

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

Before:

<img width="1504" alt="Screenshot 2024-07-23 at 1 16 24 PM" src="https://github.com/user-attachments/assets/037a79f4-8c6a-41e4-b580-17f685a2931c">

<img width="1508" alt="Screenshot 2024-07-23 at 1 16 35 PM" src="https://github.com/user-attachments/assets/9b591134-ed23-47fc-8de9-8a0bdf95fed1">

<img width="774" alt="Screenshot 2024-07-23 at 1 22 53 PM" src="https://github.com/user-attachments/assets/03cba33a-2a48-43a0-b3b0-16a040854745">

After:

<img width="1504" alt="Screenshot 2024-07-23 at 12 53 18 PM" src="https://github.com/user-attachments/assets/87bf0005-cf9e-4210-949f-48cc6c19d87b">

<img width="1506" alt="Screenshot 2024-07-23 at 12 53 32 PM" src="https://github.com/user-attachments/assets/a2402539-15af-4eae-b634-caa603c6694f">

<img width="922" alt="Screenshot 2024-07-23 at 1 23 04 PM" src="https://github.com/user-attachments/assets/86f1980f-3002-45d7-94d2-c6d69958b6b9">

- Updated after removing `formatjs` dependency:

<img width="509" alt="Screenshot 2024-08-19 at 4 18 37 PM" src="https://github.com/user-attachments/assets/50eeaa85-5d06-4720-9a7b-aea24e347418">

praveenkumar1798 reviewed (changes requested):
@rajat-ws There is a format js asset that is consuming about 1.3 Mb which is ideally not required. I have fixed it in react template. please do refer.

![image](https://github.com/user-attachments/assets/ef261148-2258-4a0c-bf7d-1151ff0093d8)

rajat-ws commented:
> @rajat-ws There is a format js asset that is consuming about 1.3 Mb which is ideally not required. I have fixed it in react template. please do refer.
> 
> ![image](https://private-user-images.githubusercontent.com/80456905/351573303-ef261148-2258-4a0c-bf7d-1151ff0093d8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwNTEwNTIsIm5iZiI6MTcyNDA1MDc1MiwicGF0aCI6Ii84MDQ1NjkwNS8zNTE1NzMzMDMtZWYyNjExNDgtMjI1OC00YTBjLWJmN2QtMTE1MWZmMDA5M2Q4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE5VDA2NTkxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmNWY4YTQ0NTg5MTRjOWY5Y2E3NzM3NTJmM2M3NjJjMDNlZGNiMmYwZmFhZDRmOGY5NTVjZTMyMmNhNWM1YTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.B8tYoJCBmj0_mhQCkUMFYhUu3HDcDKA0GlTcxSh3gVM)

Removed it.

praveenkumar1798 commented on app/themes/colors.js:
```diff
@@ -5,14 +5,16 @@
-const primary = '#000000';
-const text = '#212529';
+const primary = '#2F4858';
+const lightGreen = '#607274';
+const text = '#4D3D0C';
 const secondaryText = '#FFFFFF';
+const tertiaryText = '#4D3D0C';
 const secondary = '#b0b0b0';
 const success = '#28a745';
 const error = '#dc3545';
 const gotoStories = '#1890ff';
-const cardBg = '#d8d8d8';
+const cardBg = '#EFF1F3';
```
Why are these changes there ?

rajat-ws replied:
Current ui of app looked somewhat awful hence I tried to modify it. Have added screenshots in the PR description for the reference.

praveenkumar1798 commented on app/containers/LanguageProvider/index.tsx:
```diff
@@ -10,21 +10,22 @@ import React, { PropsWithChildren } from 'react';
-  );
+  const localizedMessages = messages[locale!];
+
+  i18n.load(locale!, localizedMessages);
+  i18n.activate(locale!);
```
Why is it locale!. Is this expectedb ?

rajat-ws replied:
Pushed the fix for it.

praveenkumar1798 commented on app/components/Siderbar/index.tsx:
```diff
@@ -35,7 +35,7 @@ const SideBarStatic = styled.div`
     min-height: calc(100vh - ${(props) => props.theme.headerHeight});
     height: auto;
-    background-color: ${colors.primary};
+    background-color: ${colors.lightGreen};
```
Why have we changed this ?

rajat-ws replied:
Tried to make UI more appealing and hence made those changes. I have added screenshots in the PR description for the reference.

praveenkumar1798 commented on app/components/Header/index.tsx:
```diff
@@ -18,7 +18,7 @@ const StyledHeader = styled(Layout.Header)`
       align-items: center;
       justify-content: center;
-      background-color: ${colors.primary};
+      background-color: ${colors.lightGreen};
```
Why have changed this ? Any reason ?

rajat-ws replied:
Tried to make UI more appealing and hence made those changes. I have added screenshots in the PR description for the reference.
